src/Apps/Order/routes.tsx

```diff
@@ -65,9 +71,45 @@ export const routes: RouteConfig[] = [
         }
         order: ecommerceOrder(id: $orderID) {
           state
+          requestedFulfillment {
+            __typename
+          }
+          creditCard {
+            id
+          }
         }
       }
     `,
+    render: ({ Component, props }) => {
+      if (Component && props) {
+        const { location, order, params } = props as any
+
+        if (
+          order &&
+          order.state !== "PENDING" &&
+          !location.pathname.includes("status")
+        ) {
+          throw new RedirectException(`/order2/${params.orderID}/status`)
+        } else if (
```

Comment:
It would be helpful if there was a short comment between each conditional branch